go/vt/mysqlctl/backup.go

```diff
@@ -315,21 +315,22 @@ func removeExistingFiles(cnf *Mycnf) error {
 
 // ShouldRestore checks whether a database with tables already exists
 // and returns whether a restore action should be performed
-func ShouldRestore(ctx context.Context, params RestoreParams) (bool, error) {
-	if params.DeleteBeforeRestore || RestoreWasInterrupted(params.Cnf) {
+func ShouldRestore(ctx context.Context, logger logutil.Logger, cnf *Mycnf, mysqld MysqlDaemon,
```

Comment:
these changes allow this to be called from both `tabletmanager.restoreFromClone()` and `tabletmanager.RestoreBackup()` without the former needing to construct `RestoreParams`, which is otherwise not needed by `restoreFromClone()`.